csharp/Microsoft.Azure.Databricks.Client/PipelinesApiClient.cs

```diff
@@ -0,0 +1,192 @@
+    }
+
+    public async Task Reset(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/reset";
+        await HttpPost(this.HttpClient, requestUri, new { }, cancellationToken);
+    }
+
+    public async Task Stop(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/stop";
+        await HttpPost(this.HttpClient, requestUri, new { }, cancellationToken);
+    }
+
+    public async Task<Pipeline> Get(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}";
+        return await HttpGet<Pipeline>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<PipelineUpdate> GetUpdate(string pipelineId, string updateId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/updates/{updateId}";
+        var response = await HttpGet<Dictionary<string, PipelineUpdate>>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+        return response["update"];
+    }
+
+    public async Task<PipelineUpdatesList> ListUpdates(
+        string pipelineId,
+        int maxResults = 25,
+        string pageToken = null,
+        string untilUpdateId = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUriSb = new StringBuilder($"{ApiVersion}/pipelines/{pipelineId}/updates?max_results={maxResults}");
+
+        if (pageToken != null)
+        {
+            requestUriSb.Append($"&page_token={pageToken}");
+        }
+
+        if (untilUpdateId != null)
+        {
+            requestUriSb.Append($"&until_update_id={untilUpdateId}");
+        }
+
+        var requestUri = requestUriSb.ToString();
+        return await HttpGet<PipelineUpdatesList>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<string> Start(
+        string pipelineId,
+        bool fullRefresh = false,
+        PipelineUpdateCause cause = PipelineUpdateCause.API_CALL,
+        string refreshSelection = default,
+        string fullRefrehSelection = default,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/updates";
+
+        var requestDict = new Dictionary<string, string>()
+        {
+            { "full_refresh", fullRefresh.ToString().ToLower() },
+            { "cause", cause.ToString() }
+        };
+
+        if (refreshSelection != null)
+        {
+            requestDict["refresh_selection"] = refreshSelection;
+        }
+
+        if (fullRefrehSelection != null)
+        {
+            requestDict["full_refresh_selection"] = fullRefrehSelection;
+        }

```

Comment:
```suggestion
        if (refreshSelection != null)
        {
            requestDict["refresh_selection"] = refreshSelection.ToList();
        }

        if (fullRefrehSelection != null)
        {
            requestDict["full_refresh_selection"] = fullRefrehSelection.ToList();
        }

```

Reply:
Please check that the request object is serialized to this format:

```json
{
   "refresh_selection": [
     "string1", "string2"
   ],
  "full_refresh_selection": [
     "string1", "string2"
  ]
}
```

Reply:
Looks good. I also added test for the request to check the format